### PR TITLE
#11 unfollow user

### PR DESCRIPTION
**Pull request by wesol05**

### CODE REVIEW

File: app/config/services.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 services:
     use_case.register_user:
-        class: AppBundle\UseCase\RegisterUserUseCase
+        class: AppBundle\UseCase\RegisterUser
```
Comment (wesol05) [inline app/config/services.yml:3]: already in UseCase namespace, so suffix is no needed IMO

File: app/config/services.yml [hunk 2]
```diff
@@ -11,12 +11,12 @@ services:
...
 
     use_case.user_follow_user:
-        class: AppBundle\UseCase\MakeUserFollowUserUseCase
+        class: AppBundle\UseCase\FollowUser
         arguments:
             - '@repository.user'
 
     repository.user:
-        class: AppBundle\Repository\UserRepository
+        class: AppBundle\Repository\DoctrineUsers
```
Comment (wesol05) [inline app/config/services.yml]: important thing is that this is doctrine implementation of UserRepository interface

File: spec/AppBundle/UseCase/RegisterUserUseCaseSpec.php [hunk 3]
```diff
@@ -3,26 +3,26 @@
 namespace spec\AppBundle\UseCase;
 
 use AppBundle\Entity\User;
-use AppBundle\UseCase\Command\RegisterUserCommand;
-use AppBundle\UseCase\RegisterUserUseCase;
-use Core\Repository\UserRepositoryInterface;
+use AppBundle\UseCase\RegisterUserCommand;
+use AppBundle\UseCase\RegisterUser;
+use Core\Repository\UserRepository;
```
Comment (wesol05) [inline spec/AppBundle/UseCase/RegisterUserUseCaseSpec.php]: ide tells us that this is interface on every step, so no need to add such suffix

Comment (matigda) [reply spec/AppBundle/UseCase/RegisterUserUseCaseSpec.php]: It's about the concept. I prefer when every interface has it's suffix.

File: src/AppBundle/UseCase/RegisterUser.php [hunk 4]
```diff
@@ -44,7 +36,7 @@ public function execute(RegisterUserCommand $command): User
             $user->setImage($command->getImage());
         }
 
-        $this->userRepository->add($user);
```
Comment (matigda) [inline src/AppBundle/UseCase/RegisterUser.php]: It's a repository, not a collection itself, so I would rather stay with "userRepository" variable.

File: app/config/services.yml (hunk 1 above)
Comment (matigda) [reply app/config/services.yml:3]: Ok.

File: spec/AppBundle/UseCase/RegisterUserUseCaseSpec.php (hunk 3 above)
Comment (wesol05) [reply spec/AppBundle/UseCase/RegisterUserUseCaseSpec.php]: ok, I will rollback this

File: src/AppBundle/UseCase/RegisterUser.php (hunk 4 above)
Comment (wesol05) [reply src/AppBundle/UseCase/RegisterUser.php]: ok 👍

File: src/AppBundle/Repository/DoctrineUsers.php [hunk 5]
```diff
@@ -7,10 +7,10 @@
 use AppBundle\ReadModel\Query\UserProfileQuery;
 use AppBundle\ReadModel\View\UserProfileView;
 use Core\Entity\User;
-use Core\Repository\UserRepositoryInterface;
+use Core\Repository\UserRepository;
 use Doctrine\ORM\EntityRepository;
 
-class UserRepository extends EntityRepository implements UserRepositoryInterface, UserProfileQuery
+class DoctrineUsers extends EntityRepository implements UserRepository, UserProfileQuery
```
Comment (wesol05) [inline src/AppBundle/Repository/DoctrineUsers.php]: what about DoctrineUserRepository? or DoctrineUsers is fine for you?

File: src/AppBundle/UseCase/UnfollowUser.php [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+use Core\Repository\UserRepositoryInterface;
+
+final class UnfollowUser
+{
+    private $users;
+
+    public function __construct(UserRepositoryInterface $users)
+    {
+        $this->users = $users;
```
Comment (matigda) [inline src/AppBundle/UseCase/UnfollowUser.php:13]: If we stick to the naming with "Interface" as a suffix and to the concept that repository != collection, then I think "$users" variable should be called "$usersRepository" instead.

Comment (wesol05) [reply src/AppBundle/UseCase/UnfollowUser.php:13]: :+1:

File: src/Core/Repository/BaseRepositoryInterface.php [hunk 7]
```diff
@@ -0,0 +1,8 @@
+<?php
+
+namespace Core\Repository;
+
+interface BaseRepositoryInterface
+{
+    public function save();
```
Comment (matigda) [inline src/Core/Repository/BaseRepositoryInterface.php:7]: As we discussed - it's either unit of work with data mapper or update($user) method. So, as it should be done the best way there is, let's try to bring another abstraction layer ;)

**Pull request merged**